DBZ-8121 Document pgvector (#5737)
debezium/debezium: documentation/modules/ROOT/pages/connectors/postgresql.adoc

|===

[id="postgresql-pgvector-types"]

> === PgVector types

Reviewer: ```suggestion
=== pgvector types
```

The PostgreSQL connector supports all link:https://github.com/pgvector/pgvector[pgvector extension data types].

> .Mappings of pgvector data types

Reviewer: ```suggestion
.Mappings of PostgreSQL `pgvector` data types
```

[cols="25%a,20%a,55%a",options="header"]
|===
|pgvector data type
|Literal type (schema type)
|Semantic type (schema name) and Notes

|`VECTOR`
|`ARRAY (FLOAT64)`
|`io.debezium.data.Vector` +

|`HALFVEC`
|`ARRAY (FLOAT32)`
|`io.debezium.data.HalfVector` +

|`SPARSEVEC`
|`STRUCT`

> |`io.debezium.data.SparseVector` +
>  +

Reviewer: ```suggestion
|`io.debezium.data.SparseVector` 
 
```

> Contains a structure with two fields: +
> 
> * `dimensions (INT16)` - Dimensions (total length of the sparse vector) of the sparse vector.
> * `vector (MAP (INT16, FLOAT64))` - A map representing the sparse vector. Each value is is a combination of index of vector element (starting with `1`) and the value of the element.

Reviewer: ```suggestion
Contains a structure that includes the following fields: 

`dimensions (INT16)`:: The total length of the sparse vector.
`vector (MAP (INT16, FLOAT64))`:: A map that represents the sparse vector.  +
Each map value includes the following elements:

* Index number of the vector element (starting with `1`).
* Value of the vector element.
```
